cmd/version.go

```diff
@@ -17,58 +17,91 @@ limitations under the License.
 package cmd
 
 import (
+	"encoding/json"
 	"fmt"
+	"runtime"
 	"runtime/debug"
 )
 
 const unknown = "unknown"
 
-// var needs to be used instead of const as ldflags is used to fill this
-// information in the release process
+// These are filled via ldflags during build
 var (
 	kubeBuilderVersion      = unknown
 	kubernetesVendorVersion = "1.33.0"
 	goos                    = unknown
 	goarch                  = unknown
-	gitCommit               = "$Format:%H$" // sha1 from git, output of $(git rev-parse HEAD)
-
-	buildDate = "1970-01-01T00:00:00Z" // build date in ISO8601 format, output of $(date -u +'%Y-%m-%dT%H:%M:%SZ')
+	gitCommit               = "$Format:%H$"
+	buildDate               = "1970-01-01T00:00:00Z"
 )
 
-// version contains all the information related to the CLI version
-type version struct {
+// VersionInfo holds all CLI version-related information
+type VersionInfo struct {
 	KubeBuilderVersion string `json:"kubeBuilderVersion"`
 	KubernetesVendor   string `json:"kubernetesVendor"`
 	GitCommit          string `json:"gitCommit"`
 	BuildDate          string `json:"buildDate"`
-	GoOs               string `json:"goOs"`
+	GoOS               string `json:"goOs"`
 	GoArch             string `json:"goArch"`
 }
 
-// versionString returns the Full CLI version
-func versionString() string {
+// resolveBuildInfo ensures dynamic fields are populated
+func resolveBuildInfo() {
 	if kubeBuilderVersion == unknown {
 		if info, ok := debug.ReadBuildInfo(); ok && info.Main.Version != "" {
 			kubeBuilderVersion = info.Main.Version
 		}
 	}
-
-	return fmt.Sprintf("Version: %#v", version{
-		kubeBuilderVersion,
-		kubernetesVendorVersion,
-		gitCommit,
-		buildDate,
-		goos,
-		goarch,
-	})
+	if goos == unknown {
+		goos = runtime.GOOS
+	}
+	if goarch == unknown {
+		goarch = runtime.GOARCH
+	}
```

Comment:
I think we shouldn't make this change.

Since all builds run on Ubuntu (see https://github.com/kubernetes-sigs/kubebuilder/blob/master/build/.goreleaser.yml), falling back to `runtime.GOOS` / `GOARCH` will always return `linux/amd64`, even for macOS targets.

These values are already injected via `-ldflags`, so if they’re missing, it’s better to leave them as `"unknown"` rather than misreporting the platform.

Let’s avoid overriding them at runtime.

Reply:
Also, see : https://github.com/kubernetes-sigs/kubebuilder/pull/4516/files 
It can either be used if kubebuilder is installed with go install module
(another behaviour that we need to consider)